base/lnls-prune-artifacts.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+set -eu
+
+
+filter_out_dirs() {
+    list="$1"
+    exclude_list="$2"
+
+    for item in $exclude_list; do
+        dir=$item
+
+        while [ "$dir" != "/" ]; do
+            list=$(echo "$list" | grep -xv $dir)
+
+            dir=$(dirname $dir)
+        done
+    done
+
+    echo "$list"
+}
+
+find_elf_executables() {
+    targets=$@
+
+    while read -r executable; do
+        read -r -N 4 trailer < "$executable"
+
+        # Output only ELF binaries
+        if [ "$trailer" = $'\x7fELF' ]; then
+            echo $executable
+        fi
+    done < <(find $targets -type f -executable)
```

Comment:
We must loop this way here because they might be file names with spaces. Indeed, MCA is such a representative of such scenario.

Because of that, I wonder if I haven't assumed to much elsewhere regarding file names.

Also, would this deserve a comment or is this clear enough?

Reply:
> Do you mean using this instead of for executable in $(find ...?

Exactly.

> You can just leave a comment on top of the loop that file names might have spaces.

Okay. Should I do the same for every other loop as well?

Reply:
> I kept this as a loop with such requirement because we mostly control such paths. Let me know what you think about this.

I agree with this justification here, especially because it makes loops more complicated.